### ISSUE TITLE
getEuclideanDistance should depends on the state of timeseries

### ISSUE THREAD

**Issue by assaad** [patrickzib/SFA#28]
The getEuclidean Distance gives incorrect results. 
It should depends wether the timeseries is normalized or not and wether the query is normalized or not. 
In case of normalization, both should be de-normalized (value * std + avg). 
The current implementation normalize again !
Cheers! 
Assaad

**Comment by patrickzib** [comment]
There are two cases:
query is z-normed: thus we have to z-norm all the subsequences.

query is not z-normed: 
a) we can either force the query to be z-normed, which is one call to query.norm() before performing the query or 
b) we can not apply z-normalization at all, which would be a check in getDistance. 

Typically, we force the query to be z-normed in this case. However, I agree that it is possible to leave out z-normalization, too. The proposed pull-request however, breaks the code by removing z-normalization for subsequences.

**Comment by assaad** [comment]
For my understanding, it's better to return the un-normalized distance measure. So a potential fix will be: if the query is normalized -> unnormalize its values, if the timeseries is normalized -> un-normalize the values -> then calculate the euclidean distance between original un-normalized versions of both timeseries data and query :) 
Big thanks anyway!

**Comment by patrickzib** [comment]
Removing it, is not sooo easy. For example, SFA uses the momentary Fourier transform. The MFT applies z-normalization to each subsequence:
https://github.com/patrickzib/SFA/blob/43cfad8696591f4a526ff8372fbbbfee161d215b/src/main/java/sfa/transformation/MFT.java#L112

The same does `TimeSeries.getDisjointSequences` and `TimeSeries.getSubsequences`

So, when removing it from `getDistance`, you suddenly try to lower bound un-normalized subsequences with z-normalized words. This should fail.

**Comment by assaad** [comment]
yes i know, that's why i didn't propose a fix. I saw that in subsequence matching, each subsequence is normalized with different mean and std, so i can't un-normalized. maybe we need 2 separate methods.
